sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+
+- The fields contained in the **robot** resource can be described as the table below:
+
+|        Fields       |   Type   |              Description               |
+| :-----------------: | :------: | :------------------------------------: |
+|       robotID       |   uint   |            Unique ID of a robot        |
+| registration_status | boolean  |  Indicates whether robot is registrated  |
+|      position       |  struct  | Indicates the location of the robot, including coordinates, the last point passed, etc. |
+|   running_status    |  struct  | Represents information related to robot motion, including linear velocity and angular velocity |
+|     task_info     |  struct  | Including the identification of orders and tasks, a task can be represented by a series of points and segments |
+|    battery_status    |  struct  | Including information such as remaining battery power, charging status, etc. |
+|   abnormal_events   | struct[] |         Record the exception information of the node         |
+|    robot_status     | boolean  |  indicates whether robot is ready for scheduling  |
+
+-  The fields contained in the task resource can be described as the table below:
+
+|          Field          |   Type   |                 Description                 |
+| :---------------------: | :------: | :-----------------------------------------: |
+|         orderID         |   uint   | The unique identification of the order in the system, used to identify whether multiple tasks belong to the same order |
+|         taskID          |   uint   | The orderID and taskID jointly identify the task status currently executed by the robot |
+|       allocated         |  string  | Empty if not assigned, otherwise the assigned robotID |
+|  point_status_sequence  | struct[] |     Sequence of points already received and not executed     |
+| segment_status_sequence | struct[] |    Sequence of segment already received and not executed     |
+| task_destination_ID  |  uint  | Describe the task destination point ID  |
+
+- The fields contained in the robotSync resource can be describe as follow:
+
+|          Field          |   Type   |             Description         |
+| :---------------------: | :------: | :-----------------------------: |
+|     registed_robots     | string[] | Keep a list of registered nodes, including nodes name and a list of node devices |
+
+## Controller design
+
+The Controller is mainly used to monitor the newly added custom resources in the cluster, or the changes of existing resources, and execute corresponding logic based on these informations.
+
+For *taskController*
+
+1. When a new task resource is created in the cluster, taskController should monitor this change
+2. The taskController match the task to a node that can execute the task
+3. Then the Controller should modify the task_info field of the node
+
+For the *robotControlelr*
+
+1. The robotController monitors the changes of the robot resources
```

Comment:
This place isn't very clear. For an edge node, we always install edgecore by `keadm join xxx`. What is the process of registering a robot and installing edgecore?